plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeSourceHandle.java

```diff
@@ -88,9 +99,92 @@ public Optional<byte[]> getSecretKey()
     public String toString()
     {
         return toStringHelper(this)
+                .add("exchangeId", exchangeId)
                 .add("partitionId", partitionId)
                 .add("files", files)
                 .add("secretKey", secretKey.map(value -> "[REDACTED]"))
                 .toString();
     }
+
+    public static class SourceFile
```

Comment:
can this be record? Or you worried about `getRetainedSizeInBytes`?

Reply:
Yeah, I didn't want to implement `getRetainedSizeInBytes` myself.